consensus/module.go

```diff
@@ -27,42 +27,43 @@ var _ modules.PacemakerConfig = &typesCons.PacemakerConfig{}
 var _ modules.ConsensusConfig = &typesCons.ConsensusConfig{}
 var _ modules.ConsensusModule = &ConsensusModule{}
 
-// TODO(olshansky): Any reason to make all of these attributes local only (i.e. not exposed outside the struct)?
-// TODO(olshansky): Look for a way to not externalize the `ConsensusModule` struct
+// TODO: Do not export the `ConsensusModule` struct or the fields inside of it.
 type ConsensusModule struct {
 	bus        modules.Bus
 	privateKey cryptoPocket.Ed25519PrivateKey
-	consCfg    modules.ConsensusConfig
+
+	consCfg     *typesCons.ConsensusConfig
+	consGenesis *typesCons.ConsensusGenesisState
 
 	// Hotstuff
 	Height uint64
 	Round  uint64
 	Step   typesCons.HotstuffStep
-	Block  *typesCons.Block // The current block being voted on prior to committing to finality
+	Block  *typesCons.Block // The current block being proposed / voted on; it has not been committed to finality
 
 	HighPrepareQC *typesCons.QuorumCertificate // Highest QC for which replica voted PRECOMMIT
 	LockedQC      *typesCons.QuorumCertificate // Highest QC for which replica voted COMMIT
 
 	// Leader Election
 	LeaderId       *typesCons.NodeId
 	NodeId         typesCons.NodeId
-	ValAddrToIdMap typesCons.ValAddrToIdMap // TODO(design): This needs to be updated every time the ValMap is modified
-	IdToValAddrMap typesCons.IdToValAddrMap // TODO(design): This needs to be updated every time the ValMap is modified
+	ValAddrToIdMap typesCons.ValAddrToIdMap // TODO: This needs to be updated every time the ValMap is modified
+	IdToValAddrMap typesCons.IdToValAddrMap // TODO: This needs to be updated every time the ValMap is modified
 
 	// Consensus State
-	appHash      string
+	lastAppHash  string // TODO: Always retrieve this variable from the persistence module and simplify this struct
 	validatorMap typesCons.ValidatorMap
 
 	// Module Dependencies
-	utilityContext    modules.UtilityContext
+	UtilityContext    modules.UtilityContext
```

Comment:
Why is this public now? As far as I can tell it's only used within consensus

Reply:
Because of the following line in `consensus/consensus_tests/pacemaker_test.go`

```go
consensusModImpl.FieldByName("UtilityContext").Set(reflect.ValueOf(utilityContext))
```

Reflect doesn't let us set the values of unexported fields and I'm utilizing the inner values of the consensus implementation to "pre-set it" to a certain state.

Some solutions to this involve using unsafe pointers, which I want to avoid, but the better solution is likely to either:
1. Move the tests into the appropriate package
2. Create an interface so we can mock the values of the consensus module

Going to think about this some more...

Reply:
I think this is going to be a non-trivial amount of work so I created #256.

Reply:
I think this can be done by:
1. creating a `module_test.go` in consensus
2. inside of it have something like this:
```go
func (c *ConsensusModule) SetUtilityContext(utilityContext modules.UtilityContext) {
	c.utilityContext = utilityContext
}
```
3. in `pacemaker_test.go` use reflection to call the set method (that's available only during tests):

```go
consensusModImpl.MethodByName("SetUtilityContext").Call([]reflect.Value{reflect.ValueOf(utilityContext)})
```

this way we can keep the `utilityContext` private

Reply:
Haha, it worked!

I was thinking of doing the same thing for `Round`, `Height`, etc., but it feels wrong to create setters for all of them solely for testing purposes. 

Since the exposure of utility context was a step backwards, I definitely like this approach better, but for the other params, I want to keep the ticket open and figure out something better.